src/content/mod_configuration.cpp

```diff
@@ -221,10 +224,19 @@ void ModConfiguration::checkConflictsAndDeps()
 
 void ModConfiguration::resolveDependencies()
 {
-	// Step 1: Compile a list of the mod names we're working with
+	// Compile a list of the mod names we're working with
 	std::set<std::string> modnames;
-	for (const ModSpec &mod : m_unsatisfied_mods) {
-		modnames.insert(mod.name);
+	std::optional<ModSpec> first_mod_spec, last_mod_spec;
+	for (ModSpec &mod : m_unsatisfied_mods) {
+		if (mod.name == m_first_mod) {
+			first_mod_spec = mod;
+		} else if (mod.name == m_last_mod) {
+			// only non optional depends have to be check for last mod
+			mod.unsatisfied_depends = mod.depends;
```

Comment:
does this actually do anything? I see this being overriden in the loop below

perhaps `mod.optdepends.clear();` would make more sense

Reply:
this seems to be unnecessary indeed; i think clearing the optional depends shouldn't be necessary either: if the opt depends are there, then they will be added to the depends, and hence processed before the last mod and satisfied.

i think the correct thing to do is to just remove the line